### ISSUE TITLE
Proposed warning: `assert <generator_expression>`

### ISSUE THREAD

**Issue by gschaffner** [PyCQA/flake8-bugbear#534]
For example:

```python
assert (result[i] == expected[i] for i in range(len(expected)))
assert (result[key] == expected[key] for key in result.keys() | expected.keys())
```

bugbear could warn and suggest adding `all()`.

Note that even an empty generator expression is truey:

```python
assert (False for _ in range(0))
```

so it's fine that bugbear doesn't statically know the length of the generator.

There are a couple of similar warnings:

*   F631 is a similar rule for `assert <tuple>`.
*   Proposal (2) in #440 is a similar rule for `assert <constant>`.

**Comment by JelleZijlstra** [comment]
Seems useful!

This could even be added to Python itself as a SyntaxWarning.

**Comment by cooperlees** [comment]
Sounds good to me too ... Thanks for the PR too. Will check it out.

**Comment by cooperlees** [comment]
Merged a partial solution - we wonder if it would make sense to broaden this into asserts with other constructs that we can statically detect will always be true, such as nonempty list literals ...
